File: crates/indexing/tests/fixtures/c/sample.c
/* Free function at file scope. */
int free_function(void) {
    int x = 1;
    return x;
}

/* Returns a greeting length for the given name. */
int documented(const char *name) {
    (void)name;
    return 0;
}

struct Tracker {
    int value;
};

void tracker_init(struct Tracker *t) {
    t->value = 0;
}

void tracker_update(struct Tracker *t) {
    t->value += 1;
}

File: crates/indexing/tests/fixtures/typescript/sample.ts
/** Free function at module scope. */
export function freeFunction(): number {
  const x = 1;
  return x;
}

/** Returns a greeting for the given name. */
export function documented(name: string): string {
  return `hello ${name}`;
}

export class Tracker {
  value: number = 0;

  constructor() {
    this.value = 0;
  }

  update(): void {
    this.value += 1;
  }
}

export function itWorks(): void {
  // test-like
}
